bsip-0062.md

```diff
@@ -0,0 +1,44 @@
+ ```
+  BSIP: 0062
+  Title: Close short position 
+  Authors:
+    Stefan Schießl stefan.schiessl@blockchainprojectsbv.com
+    Jerry Liu bitcrab@qq.com
+  Status: Draft
+  Type: Protocol
+  Created: 2019-09-14
+  Worker: TBD
+  ```
+
+# Abstract
+Closing a short position comes with a risk of getting margin called if no spare funds are available to buy the long and reduce the debt to zero. This BSIP introduces a way to put the short position directly on the market.
+
+# Rationale
+Closing a short position with a healthy CR should come with no risk of getting margin called.
+
+# Specification
+Introduce a way that user can directly place orders to sell collateral in a margin position, and the received SmartCoins will be used to reduce the debt automatically.
+
+User needs to define the amount to sell and amount to receive while placing the sell orders, and the 
```

Comment:
I recommend that this proposal should explicitly have a maximum amount (a cap) of collateral that can be offered.  The reason being that while a "close short position" remains on the books, enough collateral must be retained by the blockchain to honor any possible redemption requests by force settlements.

For example if a debt position's CR = 1.7x and the asset's MCR = 1.6x, the "close short order" should not be able to **initially** offer more than 0.1x worth of the debt (denominated in collateral asset) in an open order at a time.  Ideally the the size of the "close short order" could be dynamic to fluctuate with the valuation of the collateral; but this could be technically difficult and might be relatively computationally expensive.

Some might argue that initial amount could be 0.7x of the debt (denominated in collateral) but then this amount must certainly be dynamic because there would be no buffer for the blockchain to honor force settlements.

Reply:
That is a very important point. Thanks!

Reply:
> For example if a debt position's CR = 1.7x and the asset's MCR = 1.6x, the "close short order" should not be able to initially offer more than 0.1x worth of the debt (denominated in collateral asset) in an open order at a time.

To ensure all other mechanics work as intended it is sufficient to ensure this via the price. First come first serve also holds for Force Settlements (if this position would be indeed the one with least collateral, and the "close short position" clears it before the force settlement hits, the force settlement will simply take the next least collaterized one). Keep in mind that the force settlement offset actually benefits the margin position holder!

The ultimate restriction on the price is: `sell_price < CR * feed_price`. If it would be equal, it would not reduce your collateral ratio. The ultimate restriction on the amount is that the `receive_amount is <= debt_amount`. It would also be possible (and should be allowed!) to sell for less than feed price (making a profit while closing the position).

Reply:
> In that sense a close position order would be automatically cancelled if force-settlement hits before (close short position may not protect against force settlement, but should also not be hindered by it, same as with an open margin call).

agree, means no need to limit the close short order quantity, however when force settlements hits, it will not prohibit the force settlement from executing.

Reply:
> To ensure all other mechanics work as intended it is sufficient to ensure this via the price. First come first serve also holds for Force Settlements (if this position would be indeed the one with least collateral, and the "close short position" clears it before the force settlement hits, the force settlement will simply take the next least collaterized one). In that sense a close position order would be automatically cancelled if force-settlement hits before (close short position may not protect against force settlement, but should also not be hindered by it, same as with an open margin call).

My concern is over the proposed **dual-use** of a certain amount of an asset (in this case 1.0x worth of the debt in the form of the collateral asset): to honor force settlements, and to offer in a "close short position" order.

The proposed idea of automatic cancelling of a "close short position" if and when that debt position is force settled can financially work.  But it also make the order books _slightly misleading_ in a financial sense by making an order book appear deep with "close short position" offers _which are simultaneously backing_ potential redemption requests.

> The ultimate restriction on the price is: sell_price < CR * feed_price ...

I agree with the ideas about limiting the sell price to maintain or increase the CR

Reply:
> But it also make the order books slightly misleading in a financial sense by making an order book appear deep with "close short position" offers which are simultaneously backing potential redemption requests.

This would also be true for margin calls, right?

Reply:
> This would also be true for margin calls, right?

Touché :)

In that case the `sell_price` constraint is essential to ensure that a partial match of the "close short position" still allows the blockchain to honor possible redemption afterwards